common/src/main/java/org/opensearch/ml/common/conversational/ConversationMeta.java

```diff
@@ -0,0 +1,233 @@
+     * @param user the user who owns this conversation
+     */
+    public ConversationMeta(
+        String id,
+        Instant created,
+        Instant lastHit,
+        int numInteractions,
+        String name,
+        String user
+    ) {
+        this.id = id;
+        this.created = created;
+        this.lastHit = lastHit;
+        this.numInteractions = numInteractions;
+        this.name = name;
+        this.user = user;
+    }
+
+    /**
+     * Creates a conversationMeta object from a SearchHit object
+     * @param hit the search hit to transform into a conversationMeta object
+     * @return a new conversationMeta object representing the search hit
+     */
+    public static ConversationMeta fromSearchHit(SearchHit hit) {
+        String id = hit.getId();
+        return ConversationMeta.fromMap(id, hit.getSourceAsMap());
+    }
+
+    /**
+     * Creates a conversationMeta object from a Map of fields in the OS index
+     * @param id the conversation's id
+     * @param docFields the map of source fields
+     * @return a new conversationMeta object representing the map
+     */
+    public static ConversationMeta fromMap(String id, Map<String, Object> docFields) {
+        Instant created = Instant.parse((String) docFields.get(ConversationalIndexConstants.META_CREATED_FIELD));
+        Instant lastHit = Instant.parse((String) docFields.get(ConversationalIndexConstants.META_ENDED_FIELD));
+        int numInteractions = (int) docFields.get(ConversationalIndexConstants.META_LENGTH_FIELD);
+        String name = (String) docFields.get(ConversationalIndexConstants.META_NAME_FIELD);
+        String user = (String) docFields.get(ConversationalIndexConstants.USER_FIELD);
+        return new ConversationMeta(id, created, lastHit, numInteractions, name, user);
+    }
+
+    /**
+     * Creates a conversationMeta from a stream, given the stream was written to by 
+     * conversationMeta.writeTo
+     * @param in stream to read from
+     * @return new conversationMeta object
+     * @throws IOException if you're reading from a stream without a conversationMeta in it
+     */
+    public static ConversationMeta fromStream(StreamInput in) throws IOException {
+        String id = in.readString();
+        Instant created = in.readInstant();
+        Instant lastHit = in.readInstant();
+        int numInteractions = in.readInt();
+        String name = in.readString();
+        String user = in.readOptionalString();
+        return new ConversationMeta(id, created, lastHit, numInteractions, name, user);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(id);
+        out.writeInstant(created);
+        out.writeInstant(lastHit);
+        out.writeInt(numInteractions);
+        out.writeString(name);
+        out.writeString(user);
+    }
+
+    /**
+     * @return the unique id of this conversationMeta object
+     */
+    public String getId() {
```

Comment:
Same as above, let's keep the same style, use lombok to remove these code.

Reply:
ack. on it

Reply:
@jngz-es thanks! I've gone and lomboked all the AllArgsConsructors and (most of the) Getters. Is there anything else I can use Lombok for? I'm fairly new to it.